jdaviz/configs/specviz/plugins/line_analysis/line_analysis.py

```diff
@@ -69,59 +67,42 @@ def _on_viewer_data_changed(self, msg=None):
         msg : `glue.core.Message`
             The glue message passed to this callback method.
         """
-        self._viewer_id = self.app._viewer_item_by_reference(
-            'spectrum-viewer').get('id')
-
-        # Subsets are global and are not linked to specific viewer instances,
-        # so it's not required that we match any specific ids for that case.
-        # However, if the msg is not none, check to make sure that it's the
-        # viewer we care about.
-        if msg is not None and msg.viewer_id != self._viewer_id:
-            return
-
         viewer = self.app.get_viewer('spectrum-viewer')
 
-        self.dc_items = [layer_state.layer.label
-                         for layer_state in viewer.state.layers]
-
-    def vue_data_selected(self, event):
-        """
-        Callback method for when the user has selected data from the drop down
-        in the front-end. It is here that we actually parse and create a new
-        data object from the selected data. From this data object, unit
-        information is scraped, and the selected spectrum is stored for later
-        use in fitting.
-
-        Parameters
-        ----------
-        event : str
-            IPyWidget callback event object. In this case, represents the data
-            label of the data collection object selected by the user.
-        """
-        selected_spec = self.app.get_data_from_viewer("spectrum-viewer",
-                                                      data_label=event)
-
-        if self._units == {}:
-            self._units["x"] = str(
-                selected_spec.spectral_axis.unit)
-            self._units["y"] = str(
-                selected_spec.flux.unit)
+        try:
+            self._spectral_subsets = self.app.get_subsets_from_viewer("spectrum-viewer",
+                                                                      subset_type="spectral")
+        except ValueError:
+            pass
 
-        for label in self.dc_items:
-            if label in self.data_collection:
-                self._label_to_link = label
-                break
+        self.spectral_subset_items = ["Entire Spectrum"] + sorted(self._spectral_subsets.keys())
 
-        self._spectrum1d = selected_spec
+        self.dc_items = [layer_state.layer.label for layer_state in viewer.state.layers
+                         if layer_state.layer.label not in self.spectral_subset_items]
 
-        self._run_functions()
+        if len(self.dc_items) == 0:
+            self.selected_spectrum = ""
+            self.result_available = False
 
+    @observe("selected_subset", "selected_spectrum")
     def _run_functions(self, *args, **kwargs):
         """
         Run fitting on the initialized models, fixing any parameters marked
         as such by the user, then update the displauyed parameters with fit
         values
```

Comment:
While we're in this code, here's a small wording/documentation/clarifying change that might be subjective, so I won't block the merge on it. `_run_functions` is super vague to me in a software engineering context (my first impression was that it was intended to be a callback, which is NOT what this was at all!), so I borrowed some language from [the docs](https://jdaviz.readthedocs.io/en/latest/specviz/plugins.html#line-analysis) to inspire, hopefully, some more clear language? 

```suggestion
    def _calc_stats(self, *args, **kwargs):
        """
        Calculate fitting statistics on the initialized models, fixing any
        parameters marked as such by the user, then update the
        displayed parameters with fit values
```

Reply:
Good catch - the docstring isn't even accurate, there are no parameters that the user can fix here. Clearly I copied it from the Model Fitting plugin at some point and forgot to update it!

Reply:
Didn't want to block since this code isn't related to your fix, but, you know, while we're in this part of the code, may as well take the opportunity to tidy up 🤷‍♀️

Reply:
Re: tidy up -- Unless the code coverage is good here, I would defer. Tidying untested code may lead to new bugs.

Reply:
If it helps, I did run the change in my local environment and it didn't change any behavior I could see

Reply:
Ah, if you already have the patch, it is possible to PR it against this PR branch, if you have not done so already.

Reply:
Ok if we're willing to accommodate a PR fix, I can get started on it! @rosteen feel free to bypass this suggestion if there is a rush to get this in, but I'll get started on a PR fix for the above, with some additional clarifying points that I deferred (unless you're already working on it. Not sure if https://github.com/spacetelescope/jdaviz/pull/1012#issuecomment-1000419086 is referring to the rebase or the above?)

Reply:
I already updated it in the commit I just pushed.

Reply:
Perfect!